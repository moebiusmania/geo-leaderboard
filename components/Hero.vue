<script setup lang="ts">
import type { Season } from "~/server/api/seasons";

const { season } = defineProps<{
  season: Season;
}>();

const formatDate = (date: string) => {
  return new Date(date).toLocaleDateString("it-IT", {
    day: "2-digit",
    month: "long",
  });
};
</script>

<template>
  <header>
    <hgroup>
      <h1>Geografia - Season Pass</h1>
      <h2>Season {{ season.number }} - "{{ season.title }}"</h2>
    </hgroup>
    <!-- display the current season start date and end date formatted as dd/mm/yyyy -->
    <p>
      <span>{{ formatDate(season.start) }}</span>
      <span>-</span>
      <span>{{ formatDate(season.end) }}</span>
    </p>
  </header>
</template>

<style scoped>
header {
  font-family: var(--title-font);
  text-align: center;
  margin-bottom: 2.5rem;

  & hgroup {
    font-weight: 400;
    font-style: normal;

    display: flex;
    flex-direction: column;
    gap: 0.5rem;

    & h1 {
      font-size: 4.2rem;
      margin: 0.5rem 0 0;
      color: var(--primary);
      text-shadow: 2px 2px 0 var(--text);
      letter-spacing: 0.1rem;
    }

    & h2 {
      font-size: 3rem;
      margin: 0;
      /* color: var(--secondary); */
      /* text-shadow: 0 0 0.2rem var(--primary); */
    }
  }

  & p {
    margin-top: 0.2rem;
    font-style: italic;
    text-transform: capitalize;
    display: flex;
    justify-content: center;
    gap: 0.5rem;
    font-weight: bold;
  }
}
</style>
