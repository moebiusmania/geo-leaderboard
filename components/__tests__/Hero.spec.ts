import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Hero from "../Hero.vue";

describe("Hero", () => {
  const mockSeason = {
    number: 1,
    title: "Test Season",
  };

  it("renders properly with season data", () => {
    const wrapper = mount(Hero, {
      props: {
        season: mockSeason,
      },
    });

    // Check specific content
    expect(wrapper.text()).toContain("Geografia - Season Pass");
    expect(wrapper.text()).toContain(`Season ${mockSeason.number}`);
    expect(wrapper.text()).toContain(mockSeason.title);

    // Create snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("applies correct styling classes", () => {
    const wrapper = mount(Hero, {
      props: {
        season: mockSeason,
      },
    });

    const hgroup = wrapper.find("hgroup");
    const h1 = wrapper.find("h1");
    const h2 = wrapper.find("h2");

    expect(hgroup.exists()).toBe(true);
    expect(h1.exists()).toBe(true);
    expect(h2.exists()).toBe(true);
  });
});
