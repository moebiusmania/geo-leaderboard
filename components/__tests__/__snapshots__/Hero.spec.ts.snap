// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero > renders properly with season data 1`] = `
"<header data-v-30b0ac3d="">
  <hgroup data-v-30b0ac3d="">
    <h1 data-v-30b0ac3d="">Geografia - Season Pass</h1>
    <h2 data-v-30b0ac3d="">Season 1 - "Test Season"</h2>
  </hgroup><!-- display the current season start date and end date formatted as dd/mm/yyyy -->
  <p data-v-30b0ac3d=""><span data-v-30b0ac3d="">Invalid Date</span><span data-v-30b0ac3d="">-</span><span data-v-30b0ac3d="">Invalid Date</span></p>
</header>"
`;
